watertap/tools/parameter_sweep.py

```diff
@@ -16,13 +16,68 @@
+                        # OK
+                        'maxTimeLimit' : 1,                # Exceeded maximum time limited allowed by user
+                                                           #    but having return a feasible solution
+                        'maxIterations' : 2,               # Exceeded maximum number of iterations allowed
+                                                           #    by user (e.g., simplex iterations)
+                        'minFunctionValue' : 3,            # Found solution smaller than specified function
+                                                           #    value
+                        'minStepLength' : 4,               # Step length is smaller than specified limit
+                        'globallyOptimal' : 5,             # Found a globally optimal solution
+                        'locallyOptimal' : 6,              # Found a locally optimal solution
+                        'feasible' : 7,                    # Found a solution that is feasible
+                        'optimal' : 8,                     # Found an optimal solution
+                        'maxEvaluations' : 9,              # Exceeded maximum number of problem evaluations
+                                                           #    (e.g., branch and bound nodes)
+                        'other' : 10,                      # Other, uncategorized normal termination
+                        # WARNING
+                        'unbounded' : 11,                  # Demonstrated that problem is unbounded
+                        'infeasible' : 12,                 # Demonstrated that the problem is infeasible
+                        'infeasibleOrUnbounded' : 13,      # Problem is either infeasible or unbounded
+                        'invalidProblem' : 14,             # The problem setup or characteristics are not
+                                                           #    valid for the solver
+                        'intermediateNonInteger' : 15,     # A non-integer solution has been returned
+                        'noSolution' : 16,                 # No feasible solution found but infeasibility
+                                                           #    not proven
+                        # ERROR
+                        'solverFailure' : 17,              # Solver failed to terminate correctly
+                        'internalSolverError' : 18,        # Internal solver error
+                        'error' : 19,                      # Other errors
+                        # ABORTED
+                        'userInterrupt' : 20,              # Interrupt signal generated by user
+                        'resourceInterrupt' : 21,          # Interrupt signal in resources used by
+                                                           #    optimizer
+                        'licensingProblems' : 22,          # Problem accessing solver license
+        }
```

Comment:
I don't really understand this -- I would do:
```suggestion
from pyomo.opt import TerminationCondition as _TerminationCondition
_index_to_termination_condition = { idx : cond for idx, cond in enumerate(_TerminationCondition) }
_termination_condition_to_index = { cond : idx for idx, cond in _index_to_termination_condition.items() }
```
And just use the two dictionaries as needed.

Reply:
The class `TerminationConditionMapping` still exists, but the dictionary is created by importing `TerminationCondition` from `pyomo.opt`. I agree, I was duplicating things present within Pyomo and wasn't a fan of it at creation, but I don't know why I didn't think of this implementation in the first place. I do still believe that `TerminationConditionMapping` should remain because, its an added layer on top of `pyomo.opt.TerminationCondition`.

Reply:
I maintain there's no need for a class -- an instance of this object exists to store static data. If we really want the forward and reverse map in a single "container", we could do:
```
class _TerminationConditionMap:
    int_to_termination_condition = { idx : cond for idx, cond in enumerate(_TerminationCondition) }
    termination_condition_to_int = { cond : idx for idx, cond in enumerate(_TerminationCondition) }
```
Then to use:
```
_TerminationConditionMap.int_to_termination_condition[i]
```

Reply:
It has been simplified as seen here.
https://github.com/watertap-org/watertap/blob/666d10ab5414fd603a4003b9e34402b35acfa372/watertap/tools/parameter_sweep.py#L37

This container is also tested in the pytests.